circuits/cpp/src/aztec3/circuits/kernel/private/native_private_kernel_circuit.cpp

```diff
@@ -65,6 +68,17 @@ void initialise_end_values(PrivateInputs<NT> const& private_inputs, KernelCircui
     end.new_commitments = start.new_commitments;
     end.new_nullifiers = start.new_nullifiers;
 
+    if (first_iteration) {
```

Comment:
I feel like `update_end_values` is the more appropriate place to insert the nullifier. It's where we are effectively committing the outputs of this circuit iteration to the final state. This function is really just initialisation.

What do you think @dbanks12